src/sentry/shared_integrations/client.py

```diff
@@ -310,3 +311,71 @@ def head_cached(self, path, *args, **kwargs):
             result = self.head(path, *args, **kwargs)
             cache.set(key, result, self.cache_time)
         return result
+
+
+class BaseInternalApiClient(ApiClient):
+    integration_type = None
+
+    log_path = None
+
+    datadog_prefix = None
+
+    @cached_property
+    def logger(self):
+        return logging.getLogger(self.log_path)
+
+    @classproperty
+    def name_field(cls):
+        return u"%s_name" % cls.integration_type
+
+    @classproperty
+    def name(cls):
+        return getattr(cls, cls.name_field)
+
+    def track_response_data(self, code, span, error=None, resp=None):
```

Comment:
@NisanthanNanthakumar Can we make a mixin class for this and use it with `BaseApiClient`? No need to repeat code here.